Add pass extension

Tafeen opened the pull request:
## Description

<!-- A summary of your change. If you add a new extension or command, explain what it does. -->

## Screencast

<!-- If you add a new extension or command, include a screencast (or screenshot for straightforward changes). A good screencast will make the review much faster - especially if your extension requires registration in other services.  -->

## Checklist

- [ ] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [ ] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [ ] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [ ] I checked that files in the `assets` folder are used by the extension itself
- [ ] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer commented:
Hi @Tafeen 👋 

Could you look into the CI errors [explained here](https://github.com/raycast/extensions/actions/runs/6065926244/job/16455946769?pr=8219)

thomaslombart commented:
Also, could you write a description and add a recording or some screenshots? 😊

Tafeen commented:
I've fixed code style and added screenshots, thanks for tips ☺️

> Also, could you write a description and add a recording or some screenshots? 😊

Should i place description somewhere else than readme?

thomaslombart commented on extensions/pass/src/errors_handling/bad_password_path.tsx:
```diff
@@ -0,0 +1,16 @@
+import { ActionPanel, Action, Detail, openExtensionPreferences } from "@raycast/api";
```
Why not using a toast instead with a description? These are more appropriate for handling errors.

thomaslombart commented on extensions/pass/src/index.tsx:
```diff
@@ -0,0 +1,70 @@
+
+function LoadPassFilesList(PATH_TO_STORE: string) {
+  const [files, setFiles] = useState<passwords_path_structure[]>([{ pass_file_name: "", pass_file_path: "" }]);
+  const [error_loading, setError] = useState<boolean>(false);
```
Note that you can use `useCachedPromise` or `usePromise` from `@raycast/utils` if you don't want to bother with errors and loading stuff. This should simplify the code.

thomaslombart commented on extensions/pass/src/passwordDetails.tsx:
```diff
@@ -0,0 +1,168 @@
+              actions={
+                <ActionPanel title={val.name}>
+                  <Action.CopyToClipboard title={`Copy value of '${val.name}'`} content={val.value} />
+                  <Action
+                    title={"Copy Password"}
+                    onAction={() => CopyPassword(props)}
+                    shortcut={{ modifiers: ["ctrl"], key: "c" }}
+                  />
+                  <Action
+                    title={"Copy OTP"}
+                    onAction={() => CopyOTP(props)}
+                    shortcut={{ modifiers: ["ctrl"], key: "o" }}
+                  />
```
You could probably make use of icons here.

thomaslombart commented on extensions/pass/src/passwordDetails.tsx:
```diff
@@ -0,0 +1,168 @@
+export default function GetPasswordDetails(props: passwords_path_structure) {
+  return (
+    <ActionPanel>
+      <Action.Push
+        title={"Browse metadata"}
+        target={<PasswordMetadata pass_file_path={props.pass_file_path} pass_file_name={props.pass_file_name} />}
+        shortcut={{ modifiers: ["cmd"], key: "o" }}
+      />
+      <Action
+        title={"Copy Password"}
+        onAction={() => CopyPassword(props)}
+        shortcut={{ modifiers: ["ctrl"], key: "c" }}
+      />
+      <Action title={"Copy OTP"} onAction={() => CopyOTP(props)} shortcut={{ modifiers: ["ctrl"], key: "o" }} />
+      <Action.Push
+        shortcut={{ modifiers: ["ctrl"], key: "a" }}
+        title={"Create Pass Entry"}
+        target={<CreatePassForm />}
+      />
+      <Action
+        shortcut={{ modifiers: ["ctrl"], key: "d" }}
+        onAction={() => DeletePassword(props)}
+        title={"Delete Pass Entry"}
+      />
```
You could also use some icons here.

thomaslombart replied:
We have special styling for [destructive actions](https://developers.raycast.com/api-reference/user-interface/actions#action.style), it may be worth using it here.

Tafeen commented:
Hi @thomaslombart, thanks for taking time for review. Should be all good now.

thomaslombart commented:
Hey @Tafeen, sorry for being late on the PR review. The team has been off this week. I'll get back to your PR next week. Thanks for your patience! 😊

thomaslombart commented on extensions/pass/README.md:
Could you give more details on how to install the requirements. I have pass installed but I'm not sure what to do next. This could be useful for other users as well.

Tafeen replied:
Yup, it is a bit of work to make it all working together. I'll try to add better instructions next week as soon as it's possible 😄

Tafeen replied:
Added more descriptive documentation in latest updates

thomaslombart commented on extensions/pass/.eslintrc.json:
Note that we provide our ESLint configuration now. Feel free to run `npx ray migrate` to automatically get it.

Tafeen replied:
Should new ESlint config file appear in project directory after i run ``npx ray migrate``? I've manually removed previous one but the new one did not show up ☹️

thomaslombart replied:
It should yes. If it didn't work, please take a look at this page: https://developers.raycast.com/migration/v1.48.8

Tafeen replied:
Thanks, did update to eslint config

thomaslombart commented on extensions/pass/src/createPasswordForm.tsx:
```diff
@@ -0,0 +1,77 @@
+  );
+}
+
+function logEvent(event: Form.Event<string[] | string>) {
+  console.log(`Event '${event.type}' has happened for '${event.target.id}'. Current 'value': '${event.target.value}'`);
+}
```
Probably better to remove these since they don't seem to be useful and we don't want to log any passwords.

pernielsentikaer commented:
Are you still around @Tafeen

Tafeen commented:
@thomaslombart Hi can You take a look at latest update? Thanks 🤗

pernielsentikaer commented:
Hi @Tafeen! 👋

I'll take a look at this today. I'll get back to you if anything is missing 🙂

pernielsentikaer commented on extensions/pass/src/createPasswordForm.tsx:
```diff
@@ -0,0 +1,88 @@
+      title: "Adding Pass Entry",
+    });
+
+    cmd.stdout!.on("data", async () => {
```
Please don't use Non-null assertion

pernielsentikaer commented on extensions/pass/src/createPasswordForm.tsx:
```diff
@@ -0,0 +1,88 @@
+import { Preferences } from "./utils";
+
+const preferences = getPreferenceValues<Preferences>();
+const options: any = {
```
```suggestion
const options = {
```

pernielsentikaer commented on extensions/pass/src/createPasswordForm.tsx:
```diff
@@ -0,0 +1,88 @@
+      }
+    >
+      <Form.Description title="Create Password" text={"Create new password"} />
+      <Form.TextField id="path" title="File path" />
```
```suggestion
      <Form.TextField id="path" title="File Path" />
```

pernielsentikaer commented on extensions/pass/src/utils.tsx:
```diff
@@ -0,0 +1,4 @@
+export interface Preferences {
+  path_var: string;
+  passwords_store_path: string;
+}
```
This file can be deleted and `ExtensionPreferences` from Raycast can be used

pernielsentikaer commented on extensions/pass/src/createPasswordForm.tsx:
```diff
@@ -0,0 +1,88 @@
+import { Form, getPreferenceValues, Toast, showToast, Action, ActionPanel, useNavigation } from "@raycast/api";
+import { userInfo } from "os";
+import { exec } from "child_process";
+import { Preferences } from "./utils";
```
```suggestion
```

pernielsentikaer commented on extensions/pass/src/createPasswordForm.tsx:
```diff
@@ -0,0 +1,88 @@
+import { exec } from "child_process";
+import { Preferences } from "./utils";
+
+const preferences = getPreferenceValues<Preferences>();
```
```suggestion
const preferences = getPreferenceValues<ExtensionPreferences>();
```

pernielsentikaer reviewed (changes requested):
Hi 👋 

I have now tested your extension, and I have some feedback ready for you:

- Rename the folder `METADATA` to `metadata`

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

Tafeen commented:
Hi, ill try to push changes tomorrow 😄
